### PR TITLE
BDOG-1332

### PR DESCRIPTION
**Pull request by anshulbajpai**
Allow only .txt or .text files to be detected plain/text files
Fix HTML detection

### CODE REVIEW

File: app/services/tika/detectors/XmlDetector.scala [hunk 1]
```diff
@@ -14,26 +14,31 @@
...
+        Option(extractor.extractRootElement(input))
+          .fold(MediaType.OCTET_STREAM)(_ => MediaType.APPLICATION_XML)
+      } finally {
+        input.reset()
+      }
     }
 
-  }
+  private def filenameHasHtmlExtension(metadata: Metadata): Boolean =
+    Option(metadata.get(TikaMetadataKeys.RESOURCE_NAME_KEY)).exists(_.toLowerCase.matches(".*\\.htm(l)?"))
```
Comment (cjwebb) [inline app/services/tika/detectors/XmlDetector.scala:43]: What would XHTML get detected as?

Comment (anshulbajpai) [reply app/services/tika/detectors/XmlDetector.scala:43]: As `application/xhtml+xml`. Added a test for it.

**Review by cjwebb — DISMISSED** [review]

**Review by cjwebb — APPROVED** [review]

**Comment by cjwebb** [comment]
I have a feeling that Either[Something, Unit] could be done with a better pattern somehow. I just can't think how at the moment :)

**Comment by anshulbajpai** [comment]
> I have a feeling that Either[Something, Unit] could be done with a better pattern somehow. I just can't think how at the moment :)

I had the same feeling :) I could have used an ADT but then I just wanted to return the file extension in error case so the other component can do appropriate logging.

**Pull request merged**